src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionOperatorsTests.cs

```diff
@@ -0,0 +1,1722 @@
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void Increment_003_Declaration([CombinatorialValues("++", "--")] string op)
+        {
+            var src = $$$"""
+static class Extensions1
+{
+    extension(S1)
+    {
+        public void operator {{{op}}}() {}
+        public static S1 operator {{{op}}}(S1 x) => default;
+    }
+}
+
+struct S1
+{}
+""";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net90);
+
+            // PROTOTYPE: We probably should enable this scenario
+            comp.VerifyDiagnostics(
+                // (6,35): error CS0111: Type 'Extensions1' already defines a member called 'op_Increment' with the same parameter types
+                //         public static S1 operator ++(S1 x) => default;
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, op).WithArguments(UnaryOperatorName(op), "Extensions1").WithLocation(6, 35)
+                );
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void Increment_005_Declaration([CombinatorialValues("++", "--")] string op)
+        {
+            var src = $$$"""
+static class Extensions1
+{
+    extension(S1)
+    {
+        public void operator checked {{{op}}}() {}
+        public void operator {{{op}}}() {}
+    }
+}
+
+static class Extensions2
+{
+    extension(S1)
+    {
+#line 100
+        public void operator checked {{{op}}}() {}
+    }
+    extension(S1)
+    {
+        public void operator {{{op}}}() {}
+    }
+}
+
+static class Extensions3
+{
+    extension(S1)
+    {
+        public void operator checked {{{op}}}() {}
+    }
+}
+
+struct S1
+{}
+""";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net90);
+            comp.VerifyDiagnostics(
+                // PROTOTYPE: Should the pair check be performed across all extension blocks?
+
+                // (100,38): error CS9025: The operator 'Extensions2.extension(S1).operator checked ++()' requires a matching non-checked version of the operator to also be defined
+                //         public void operator checked ++() {}
+                Diagnostic(ErrorCode.ERR_CheckedOperatorNeedsMatch, op).WithArguments("Extensions2.extension(S1).operator checked " + op + "()").WithLocation(100, 38),
+
+                // (112,38): error CS9025: The operator 'Extensions3.extension(S1).operator checked ++()' requires a matching non-checked version of the operator to also be defined
+                //         public void operator checked ++() {}
+                Diagnostic(ErrorCode.ERR_CheckedOperatorNeedsMatch, op).WithArguments("Extensions3.extension(S1).operator checked " + op + "()").WithLocation(112, 38)
+                );
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void Increment_004_Declaration([CombinatorialValues("++", "--")] string op, [CombinatorialValues("virtual", "abstract", "new", "override", "sealed", "readonly", "extern")] string modifier)
```

Comment:
Consider adding a test for "unsafe" too

Reply:
> Consider adding a test for "unsafe" too

I was mostly focused on blocking modifiers that shouldn't be supported. Adding a prototype comment.